### PR TITLE
Make pipeline readable

### PR DESCRIPTION
**Pull request by nfalco79**
Hi Oleg, could you please give your opinion about the naming convention adopted to be readable pipeline?

- I use **xunit** name for the simple step like a normal plugin does
- For publisher I would mark it with prefix or postfix (better **xunit**Publisher or publishXUnit?)
- On test type I have add two names, but I would keep just one. It's better lowercase test tool name or just use the real tools name (no camel case or other kinds of convention)?

Test case will comes later when naming convention will be approved

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/xunit/XUnitWorkflowTest.java [hunk 1]
```diff
@@ -51,7 +52,7 @@ public void xunitBuilderWorkflowStepTest() throws Exception {
         job.setDefinition(new CpsFlowDefinition(""
                 + "node {\n"
                 + "  step([$class: 'XUnitBuilder', testTimeMargin: '3000', thresholdMode: 1, thresholds: [[$class: 'FailedThreshold', failureNewThreshold: '', failureThreshold: '', unstableNewThreshold: '', unstableThreshold: '1'], [$class: 'SkippedThreshold', failureNewThreshold: '', failureThreshold: '', unstableNewThreshold: '', unstableThreshold: '']], tools: [[$class: 'GoogleTestType', deleteOutputFiles: false, failIfNotNew: false, pattern: 'input.xml', skipNoTestFiles: false, stopProcessingIfError: true]]])\n"
-                + "}"));
+                + "}", true));
```
Comment (oleg-nenashev) [inline src/test/java/org/jenkinsci/plugins/xunit/XUnitWorkflowTest.java]: were you planning to update the test to use new Symbol aliases?

**Comment by nfalco79** [comment]
I had thought about it but looking the plugins installed in our Jenkins instance, some use a prefix or suffix for a Recorder or Publisher (eg: gitPublisher, publishHTML, sshPublish, publishBuildInfo), others not (eg: archiveArtifacts, cobertura, junit).
In this plugin both are present, if I were a user I would think that xunit refers to a step rather than to the publisher. In fact I do not understand the purpose of creating a simple build step, perhaps to stop the build without it being necessary to get to the post build action phase when the pipelines did not yet exist.

**Comment by nfalco79** [comment]
What about tools type?

```
xunit(thresholdMode: 1, thresholds: [failed(unstableThreshold: '1')], tools: [
	boosttest(failIfNotNew: false, pattern: '**/*.xml')
	cppunit(pattern: '**/*.xml')
])
```
or
```
xunit(thresholdMode: 1, thresholds: [failed(unstableThreshold: '1')], tools: [
	BoostTest(failIfNotNew: false, pattern: '**/*.xml')
	CppUnit(pattern: '**/*.xml')
])
```

?

**Comment by nfalco79** [comment]
@oleg-nenashev could you give me a hand to understand why the deserialization test XUnitSerialisationTest  does not works?
The main problem seems to be that the JenkinsRule does not trigger any `@Initializer` present in the current plugin. The Initializer (present since 2011) add aliases for publisher class and a shortname for all TestType extensions. I haven't found a way to inject aliases before JenkinsRule read the job configuration.

Tried cases:

1. `@JenkinsRule`+`@LocalData` -> the JenkinsRule does not trigger initializer, read the job configuration using local data but the readed project is not correctly  deserialize because miss aliases needed to deserialise correctly some nested objects.
2. `@JenkinsRule`+`@LocalData`+ call the `@Initializer` in the test method -> aliases are added but it's too late. That I remember I had try to reload configuration method and createProjectFromXML but some kind of XSTREAM cache was used.
3. `@JenkinsRule`+call the `@Initializer` in a method annotated `@Before`+`@LocalData` -> TestType aliases are gathered using rule.jenkins.getDescriptorByType but no one is available at this time, this means that some aliases are missing.
4. call the `@Initializer` in a method annotated `@BeforeClass` -> no active jenkins instance available
5. `@JenkinsRule`+call the `@Initializer` in a method annotated `@Before` -> use createProjectFromXML seems it does not use the XSTREAM serialisation (break point to ClassAliasingMapper#realClass())

**Comment by nfalco79** [comment]
I'm sorry I did see you mark this to review one hour ago otherwise I would not have final squashes.
To make unit test works I do a very dirty workaround. I add the needed alias in the `@BeforeClass` method. Maybe there is a bug on how JenkinsRule manage `@Initialiser`.

File: src/test/resources/org/jenkinsci/plugins/xunit/XUnitSerialisationTest/publisher_1_103/jobs/foo/config.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (nfalco79) [inline src/test/resources/org/jenkinsci/plugins/xunit/XUnitSerialisationTest/publisher_1_103/jobs/foo/config.xml]: remove this unused file

**Comment by oleg-nenashev** [comment]
@nfalco79 No worries. Sorry for not responding timely, was handling other things.
Do you still need my advice here? BTW, better to Cc `jenkinsci/code-reviewers` to avoid single-person bottlenecks

**Pull request merged**